src/components/Snowflake.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from "react";
+import useData from "@hooks/useData";
+import _ from "lodash";
+import { useTranslation } from "react-i18next";
+
+const Snowflake = (props) => { // if translation is missing, nothing will be loaded
+  const { t } = useTranslation();
+  const [, setData] = useData();
+
+  useEffect(() => {
+    const randomize = () => {
+      const subjects = t("campaign:letter:subject", "");
+      const subject = _.sample(subjects.split("- ").filter(m => m.length > 0));
+      let letter = [];
+      let i = 1;
+      while (t(`campaign:letter:part-${i}`, "end") !== "end") {
```

Comment:
smart! However, whenever in coding you are introducing a "special value" it is worthwhile to make it really special, so the chance someone enters "end" as a paragraph by mistake is minimal. Why not "%%END%%" ? or for example even a non-visible character, like "\3" (byte value 3 meaning: End of Text inspiration: https://www.asciitable.com/)